backend/telemetry_core/src/state/counter.rs

```diff
@@ -27,6 +30,10 @@ pub struct Counter<K> {
 
     /// The number of occurrences where the key is `None`.
     empty: u64,
+
+    /// The map of Node Id ( Network Id) to the node detail element
+    /// i.e {"123DE": "aarch64"}
+    node_map: HashMap<ArrayString<64>,K>
```

Comment:
What's the reasoning behind adding this?

As far as I understand, the relevant properties are exposed in `NodeDetails`, and then they are sent to the UI already (some depending on `--expose-node-details`). So I wonder whether you can avoid modifying all of this chain stats/counter/ranking stuff? Offhand I can't see it being used anywhere in the frontend changes.